dhcpv4/nclient4/ipv4.go

```diff
@@ -0,0 +1,376 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
```

Comment:
this makes me uncomfortable, I'd rather avoid pulling Apache code owned by Google if all we need is just IP/UDP header encoding and decoding.

For similar reasons, weeks ago I have been writing an extremely simplified IPv4/IPv6/UDP/ICMP library, which is still in a non-merged branch of another project. The code is ugly, probably overcomplicated, and definitely work-in-progress, but I have to clean it up and it may be an easier option from a ownership and licensing perspective. Should we use it instead?

The code (shame on me) is currently here, https://github.com/insomniacslk/dublin-traceroute/tree/v6/go/dublintraceroute/net . It is not as simple as the code below but we can improve it.

Reply:
as discussed in person, let's try the BSD-licensed net library first